src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -703,6 +754,84 @@ private static bool CanTypeBeExtendedPubliclyImpl(ITypeSymbol type)
                         m => m.DeclaredAccessibility != Accessibility.Internal && m.DeclaredAccessibility != Accessibility.Private && m.DeclaredAccessibility != Accessibility.ProtectedAndInternal
                     );
             }
+
+            /// <summary>
+            /// Various Visit* methods return true if an oblivious reference type is detected.
+            /// </summary>
+            private sealed class ObliviousDetector : SymbolVisitor<bool>
+            {
+                private bool _isTypeDeclaration = true;
+
+                public override bool VisitField(IFieldSymbol symbol)
+                {
+                    _isTypeDeclaration = false;
+                    return Visit(symbol.Type);
+                }
+
+                public override bool VisitMethod(IMethodSymbol symbol)
+                {
+                    _isTypeDeclaration = false;
```

Comment:
❔ Why are we not restoring this on return? What about delegating to a different visitor instance for descent into these?

Reply:
Good idea. Thanks!